molecules/main.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+import dill as pickle
+import os
+import subprocess
+import sys
+import tempfile
+
+from datetime import datetime
+
+import apache_beam as beam
+import tensorflow as tf
+import tensorflow_transform as tft
+
+from apache_beam.io import filebasedsource
+from apache_beam.options import pipeline_options
+from tensorflow_transform.tf_metadata import dataset_schema
+
+import data_extractor
+import preprocess
+import pubchem
+import trainer
+
+
+DEFAULT_WORK_DIR = os.path.join(
+    tempfile.gettempdir(), 'cloudml-samples', 'molecules')
+
+
+INPUT_SCHEMA = {
+    # Features
+    'TotalC': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalH': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalO': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalN': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+
+    # Labels
+    'Energy': dataset_schema.ColumnSchema(
+        tf.float32, [], dataset_schema.FixedColumnRepresentation()),
+}
+
+LABELS = ['Energy']
+
+
+class PubChemSource(filebasedsource.FileBasedSource):
+  def read_records(self, file_name, range_tracker):
+    with self.open_file(file_name) as f:
+      f.seek(range_tracker.start_position() or 0)
+      while range_tracker.try_claim(f.tell()):
+        for elem in pubchem.parse_molecules(f):
+          yield elem
+
+
+class PubChemToInputData(beam.PTransform):
+  def __init__(self, data_files_pattern):
+    super(PubChemToInputData, self).__init__()
+    self.data_files_pattern = data_files_pattern
+
+  def expand(self, p):
+    def count_atom_type(elem, atom_type):
+      return sum(int(atom['type'] == atom_type) for atom in elem['atoms'])
+
+    def format_molecule(elem):
+      label = float(elem['energy'])
+      return {
+        'TotalC': count_atom_type(elem, 'C'),
+        'TotalH': count_atom_type(elem, 'H'),
+        'TotalO': count_atom_type(elem, 'O'),
+        'TotalN': count_atom_type(elem, 'N'),
+        'Energy': label,
+      }
```

Comment:
Is this the same format as used in the Kaggle competition mentioned in the README?